test/combinations.jl

```diff
@@ -1,429 +1,365 @@
+            end
+        end
 
-@testitem "Meshes.Ball 2D" setup=[Setup] begin
+        iter_rules = (
+            (supports.gausskronrod, GaussKronrod()),
+            (supports.gausslegendre, GaussLegendre(100)),
+            (supports.hadaptivecubature, HAdaptiveCubature())
+        )
+
+        # Test rules
+        for (supported, rule) in iter_rules
+            if supported
+                # Scalar integrand
+                sol = testable.solution
+                @test integral(testable.integrand, testable.geometry, rule)≈sol rtol=rtol
+
+                # Callable integrand
+                f = Callable(testable.integrand)
+                @test integral(f, testable.geometry, rule)≈sol rtol=rtol
+
+                # Vector integrand
+                fv(p) = fill(testable.integrand(p), 3)
+                sol_v = fill(testable.solution, 3)
+                @test integral(fv, testable.geometry, rule)≈sol_v rtol=rtol
+            else
+                f = testable.integrand
+                geometry = testable.geometry
+                @test_throws "not supported" integral(f, geometry, rule)
+            end
+        end # for
+
+        #=
+        iter_diff_methods = (
+            (supports.autoenzyme, AutoEnzyme()),
+        )
+
+        for (supported, diff_method) in iter_diff_methods
+            @test integral(testable.integrand, testable.geometry; diff_method=diff_method)≈sol rtol=rtol
+        end
+        =#
+    end # function
+end #testsnippet
+
+#===============================================================================
+                         Create and Test Geometries
+===============================================================================#
+
+@testitem "Meshes.Ball 2D" setup=[Combinations] begin
+    # Geometry
     origin = Point(0, 0)
     radius = 2.8
     ball = Ball(origin, radius)
 
-    function f(p::P) where {P <: Meshes.Point}
+    # Integrand & Solution
+    function integrand(p::Meshes.Point)
         r = ustrip(u"m", norm(to(p)))
-        exp(-r^2)
+        exp(-r^2) * u"A"
     end
-    fv(p) = fill(f(p), 3)
-
-    # Scalar integrand
-    sol = (π - π * exp(-radius^2)) * u"m^2"
-    @test integral(f, ball, GaussLegendre(100)) ≈ sol
-    @test integral(f, ball, GaussKronrod()) ≈ sol
-    @test integral(f, ball, HAdaptiveCubature()) ≈ sol
-
-    # Vector integrand
-    vsol = fill(sol, 3)
-    @test integral(fv, ball, GaussLegendre(100)) ≈ vsol
-    @test integral(fv, ball, GaussKronrod()) ≈ vsol
-    @test integral(fv, ball, HAdaptiveCubature()) ≈ vsol
-
-    # Integral aliases
-    @test_throws "not supported" lineintegral(f, ball)
-    @test surfaceintegral(f, ball) ≈ sol
-    @test_throws "not supported" volumeintegral(f, ball)
+    solution = (π - π * exp(-radius^2)) * u"A*m^2"
+
+    # Package and run tests
+    testable = TestableGeometry(integrand, ball, solution)
+    runtests(testable, SupportStatus(:surface))
```

Comment:
```suggestion
    runtests(testable, SupportStatus(ball))
```
If we want to apply my suggestion from above. This would need to be done for each geometry.

Reply:
I just commit a change for all of the `@testitem`s. Since the geometry is stored inside of `testable` already, I made the second arg optional in `runtests(testable::TestableGeometry, supports::SupportStatus = SupportStatus(testable.geometry))' and just dropped the second arg of all calls.

Reply:
Even better. Thanks.